eras/babbage/impl/src/Cardano/Ledger/Babbage/PParams.hs

```diff
@@ -0,0 +1,474 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+-- | This module contains just the type of protocol parameters.
+module Cardano.Ledger.Babbage.PParams
+  ( PParams' (..),
+    PParams,
+    emptyPParams,
+    PParamsUpdate,
+    emptyPParamsUpdate,
+    updatePParams,
+    getLanguageView,
+    LangDepView (..),
+    encodeLangViews,
+    retractPP,
+    extendPP,
+  )
+where
+
+import Cardano.Binary
+  ( Encoding,
+    FromCBOR (..),
+    ToCBOR (..),
+  )
+import Cardano.Ledger.Alonzo.Language (Language (..))
+import Cardano.Ledger.Alonzo.PParams (LangDepView (..), encodeLangViews, getLanguageView)
+import Cardano.Ledger.Alonzo.Scripts
+  ( CostModel (..),
+    ExUnits (..),
+    Prices (..),
+    decodeCostModelMap,
+  )
+import Cardano.Ledger.BaseTypes
+  ( NonNegativeInterval,
+    Nonce,
+    StrictMaybe (..),
+    UnitInterval,
+    fromSMaybe,
+    isSNothing,
+  )
+import qualified Cardano.Ledger.BaseTypes as BT (ProtVer (..))
+import Cardano.Ledger.Coin (Coin (..))
+import Cardano.Ledger.Era
+import Cardano.Ledger.Serialization
+  ( FromCBORGroup (..),
+    ToCBORGroup (..),
+    mapToCBOR,
+  )
+import Cardano.Ledger.Shelley.Orphans ()
+import Cardano.Ledger.Shelley.PParams (HKD)
+import qualified Cardano.Ledger.Shelley.PParams as Shelley (PParams' (..))
+import Cardano.Ledger.Slot (EpochNo (..))
+import Control.DeepSeq (NFData)
+import Data.Coders
+  ( Decode (..),
+    Density (..),
+    Encode (..),
+    Field (..),
+    Wrapped (..),
+    decode,
+    encode,
+    field,
+    mapEncode,
+    (!>),
+    (<!),
+  )
+import Data.Default (Default (..))
+import Data.Functor.Identity (Identity (..))
+import Data.Map.Strict (Map)
+import GHC.Generics (Generic)
+import GHC.Records (HasField (..))
+import NoThunks.Class (NoThunks (..))
+import Numeric.Natural (Natural)
+
+type PParamsUpdate era = PParams' StrictMaybe era
+
+-- | Protocol parameters.
+-- Alonzo parameters without d and extraEntropy
+data PParams' f era = PParams
```

Comment:
@JaredCorduan, this might have been asked before, but why do we need `era` parameter here? It isn't used in any of the fields

Reply:
@lehins that's a good point, I don't see any reason for it (and similarly for the previous eras). I don't know if this is vestigial, or if we just originally era-parameterized all the types by convention.

Reply:
Very good, I suggest we remove it, at least for Babbage for sure.

Reply:
sounds good to me

Reply:
ah, this is why
![image](https://user-images.githubusercontent.com/24906808/148443568-9683e956-98d5-4777-8bdb-e8a9117e33df.png)
> If a LHS type variable that is declared as injective is not mentioned on
injective position in the RHS we conclude that the type family is not
injective.  By "injective position" we mean argument to a type constructor or
argument to a type family on injective position.
```hs
type family PParams era = (r :: Type) | r -> era
```
i don't think the injectivity annotation is strictly necessary either